porcupine/plugins/find.py

```diff
@@ -285,41 +294,41 @@ def _go_to_next_match(self, junk: object = None) -> None:
             # find first pair that starts after the cursor, or cycle back to first
             possible_indexes = (
                 i
-                for i, (start, end) in enumerate(pairs)
-                if self._textwidget.compare(start, operator, "insert")
+                for i, tag in enumerate(tags)
+                if self._textwidget.compare(f"{tag}.first", operator, "insert")
             )
             index = next(possible_indexes, 0)
-            self._select_match(pairs, index)
+            self._select_match(tags, index)
 
     def _go_to_previous_match(self, junk: object = None) -> None:
-        pairs = self.get_match_ranges()
-        if pairs:
+        tags = self.get_match_tags()
+        if tags:
             possible_indexes = (
                 i
-                for i, (start, end) in reversed(list(enumerate(pairs)))
-                if self._textwidget.compare(start, "<", "insert")
+                for i, tag in reversed(list(enumerate(tags)))
+                if self._textwidget.compare(f"{tag}.first", "<", "insert")
             )
-            index = next(possible_indexes, len(pairs) - 1)
-            self._select_match(pairs, index)
+            index = next(possible_indexes, len(tags) - 1)
+            self._select_match(tags, index)
```

Comment:
This would be cleaner with `tags.reverse()` and then the rest shared with `_go_to_next_match()`

Reply:
won't work easily, index needed for gui